File: src/Algorithms/Implementation/Java/AppleAndOrange.java
package Algorithms.Implementation.Java;

public class AppleAndOrange {

    public static void main(String[] args) {
        int[] apples = {-2, 2, 1};
        int[] oranges = {5, -6};

        int s = 7;
        int t = 11;

        int a = 5; //Larry's position
        int b = 15; //Bob's position

        int m = apples.length;
        int n = oranges.length;

        countApplesAndOranges(s, t, a, b, apples, oranges);


    }

    static void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges) {

        int landing;
        int bobPoints = 0;
        int larryPoints = 0;

        for (int i = 0; i < apples.length; i++) {
            landing = a + apples[i];
            if (landing >= s && landing <= t) 
                bobPoints++;
        }

        for (int i = 0; i < oranges.length; i++) {
            landing = b + oranges[i];

            if (landing >= s && landing <= t) 
                larryPoints++;
        }

        System.out.println(bobPoints );
        System.out.println( larryPoints);

    }
}

File: src/Algorithms/Implementation/Kotlin/AppleAndOrange.kt
package Algorithms.Implementation.Kotlin

import java.util.*

/*
 * Complete the countApplesAndOranges function below.
 */
fun countApplesAndOranges(s: Int, t: Int, a: Int, b: Int, apples: Array<Int>, oranges: Array<Int>): Unit {


    var landing: Int
    var bobPoints = 0
    var larryPoints = 0

    for (i in apples.indices) {
        landing = a + apples[i]
        if (landing in s..t)
            bobPoints++
    }

    for (i in oranges.indices) {
        landing = b + oranges[i]
        if (landing in s..t)
            larryPoints++
    }

    print(bobPoints)
    print(larryPoints)

}


fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)

    val st = scan.nextLine().split(" ")

    val s = st[0].trim().toInt()

    val t = st[1].trim().toInt()

    val ab = scan.nextLine().split(" ")

    val a = ab[0].trim().toInt()

    val b = ab[1].trim().toInt()

    val mn = scan.nextLine().split(" ")

    val m = mn[0].trim().toInt()

    val n = mn[1].trim().toInt()

    val apple = scan.nextLine().split(" ").map { it.trim().toInt() }.toTypedArray()

    val orange = scan.nextLine().split(" ").map { it.trim().toInt() }.toTypedArray()

    countApplesAndOranges(s, t, a, b, apple, orange)
}